File: PlayerDemo/AUIVideoCommon/skeleton/src/main/java/com/ethanhua/skeleton/SkeletonRecyclerView.java
package com.ethanhua.skeleton;

import android.content.Context;
import android.util.AttributeSet;

import androidx.recyclerview.widget.RecyclerView;

public class SkeletonRecyclerView extends RecyclerView {
    private SkeletonScreen skeletonScreen = null;
    private boolean mSkeletonShow = false;

    public SkeletonRecyclerView(Context context) {
        super(context);
    }

    public SkeletonRecyclerView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public SkeletonRecyclerView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public void showSkeleton(int mItemLayoutId, RecyclerView.Adapter<? extends RecyclerView.ViewHolder> mAdapter, int count) {
        skeletonScreen = Skeleton.bind(this)
                .adapter(mAdapter)
                .shimmer(false)
                .angle(20)
                .frozen(false)
                .duration(1200)
                .count(count)
                .load(mItemLayoutId)
                .show();
        mSkeletonShow = true;
    }

    public void hideSkeleton() {
        if (mSkeletonShow) {
            if (skeletonScreen != null) {
                skeletonScreen.hide();
            }
            mSkeletonShow = false;
        }
    }

    // Getters and setters for the skeletonScreen and mSkeletonShow if needed
}

File: PlayerDemo/AUIVideoCommon/common/src/main/java/com/aliyun/video/common/ui/SkeletonRecyclerView.kt
package com.aliyun.video.common.ui

import android.content.Context
import android.util.AttributeSet
import androidx.recyclerview.widget.RecyclerView
import com.ethanhua.skeleton.Skeleton
import com.ethanhua.skeleton.SkeletonScreen

/**
 * 支持框架式加载动画的 RecyclerView
 */
class SkeletonRecyclerView @kotlin.jvm.JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) :
    RecyclerView(context, attrs, defStyleAttr) {
    var skeletonScreen: SkeletonScreen? = null
    private var mSkeletonShow = false

    fun showSkeleton(mItemLayoutId: Int, mAdapter: Adapter<out ViewHolder>, count: Int) {
        skeletonScreen = Skeleton.bind(this)
            .adapter(mAdapter)
            .shimmer(false)
            .angle(20)
            .frozen(false)
            .duration(1200)
            .count(count)
            .load(mItemLayoutId)
            .show()
        mSkeletonShow = true
    }

    fun hideSkeleton() {
        if (mSkeletonShow) {
            skeletonScreen?.hide()
            mSkeletonShow = false
        }
    }
}